scripts/process/process.discharge_sparks.R

```diff
@@ -0,0 +1,102 @@
+grab_spark <- function(vals){
+  
+  xs <- seq_len(length(vals))
+  xs <- c(head(xs, 1), xs, tail(xs, 1))
+  ys <- c(min(vals), vals, min(vals))
+  x = svglite::xmlSVG({
+    par(omi=c(0,0,0,0), mai=c(0,0,0,0))
+    plot(xs, ys, type='l', axes=F, ann=F)
+  }, height=0.4, width=2)
+  xml2::xml_attr(xml2::xml_find_first(x, '//*[local-name()="polyline"]'),'points')
+}
+
+
+# Should rename at some point. Not always going to be discharge.
+process.discharge_sparks <- function(viz = as.viz('discharge-sparks')){
+  library(dplyr)
+  depends <- readDepends(viz)
+  checkRequired(depends, c("storm-gage-height", "nwis-sites"))
+  
+  sites <- depends[["nwis-sites"]]
+  gage_data <- depends[["storm-gage-height"]]
+  
+  site.nos <- names(gage_data)
+  
+  sparks <- data.frame(points = sapply(gage_data, function(x) grab_spark(x$y), USE.NAMES = FALSE),
+                       site_no = site.nos, stringsAsFactors = FALSE) %>% 
+    mutate(class = "sparkline", 
+           id = sprintf("sparkline-%s", site_no), 
+           style = "mask: url(#spark-opacity);",
+           onmouseover=sprintf("setBold('nwis-%s');", site_no), 
+           onmouseout=sprintf("setNormal('nwis-%s');hovertext(' ');", site_no),
+           onclick=sprintf("openNWIS('%s', evt);", site_no),
+           onmousemove=sprintf("hovertext('USGS %s',evt);", site_no))
+  
+  sites <- sites[c("site_no", "dec_lat_va")]
+  
+  sparks <- sparks %>% 
+    left_join(sites, by = "site_no") %>%
+    arrange(desc(dec_lat_va)) %>%
+    select(-site_no, -dec_lat_va) 
```

Comment:
Cool! My dec_lat_va came over from Irma!

Reply:
yeah, but since PR is such a fat rectangle...we may want to consider some other orientation for the sparklines....but that would be down the road a bit

Reply:
I think we should orient left to right -> bottom to top. So the earlier part of the storm shows up closer to the bottom. But it is going to be a very quick pass through compared to Irma, so it matters less

Reply:
OK, will do...

Reply:
" left to right -> bottom to top." or " right to left -> bottom to top." ?